### PR TITLE
setup ci

### PR DESCRIPTION
**Pull request by fenollp**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        toolchain: stable
+    - run: rustup component add clippy
+    - run: sudo apt install libxcb-composite0-dev
+    - uses: actions/checkout@v2
+    - uses: actions-rs/cargo@v1
+      with:
+        command: clippy
+        args: --workspace --all-targets --all-features --  -D warnings -W clippy::all
+
+  check:
```
Comment (setzer22) [inline .github/workflows/ci.yml:53]: Do we need both `check` and `clippy`? I would assume one implies the other :thinking:

**Review by setzer22 — COMMENTED** [review]
I've been meaning to add this for a while, thanks for the help! :smile: 

Looks good to me, just a minor comment

File: .github/workflows/ci.yml (hunk 1 above)
Comment (fenollp) [reply .github/workflows/ci.yml:53]: They are not the same things, no.

https://users.rust-lang.org/t/clippy-vs-cargo-check/52579
> cargo clippy runs Clippy, while cargo check only tries to compile with rustc. cargo clippy should give all errors and warnings of cargo check combined with Clippy specific warnings. Clippy warnings are a little more trigger happy and are more opinionated. The extra lint passes presumably also make it slower to run Clippy.

Comment (setzer22) [reply .github/workflows/ci.yml:53]: > cargo clippy should give all errors and warnings of cargo check combined with Clippy specific warnings

But that's exactly what I meant. Can't we just run cargo clippy? Maybe I'm missing something :sweat_smile:

Comment (fenollp) [reply .github/workflows/ci.yml:53]: Oh! Well you're right I didn't understand this properly! I'll save the extra check.

**Review by setzer22 — APPROVED** [review]
Looks good to me now. Again, thanks for working on this! :smile:

**Pull request merged**